view/frontend/web/js/components/BaseButton.vue

```diff
@@ -1,22 +1,53 @@
 <template>
-  <button :type="buttonType" :title="buttonTitle">
-    {{ text }}
+  <button
+    class="button"
+    :type="buttonType"
+    :title="buttonTitle"
+    :disabled="isBlocked"
+  >
+    <div class="button__handler">
+      <span class="button__text">
+        {{ text }}
+      </span>
+
+      <template v-if="withLoader">
+        <BaseLoader/>
+      </template>
+    </div>
   </button>
 </template>
 
 
 <script>
+import BaseLoader from './BaseLoader.vue'
+
 export default {
-  props: {
-    buttonType: {
-      type: String
-    },
-    text: {
-      type: String
-    },
-    buttonTitle: {
-      type: String
+  components: {
+    BaseLoader
+  },
+  props: ['buttonType', 'text', 'buttonTitle', 'withLoader'],
```

Comment:
Why did you change props? Prev version was ok. We should add require property.

Reply:
Maybe it would be better to cahnge prop `withLoader` to `loader`

Reply:
I think prop `loader` means it will enable loader or start the loader. In this situation, `withLoader` means it will show the button with loader option. You can also show normal button without it

Reply:
And I changed props b/c prev we just have `type: String` without require property

Reply:
So better is to add `require` true or false

Reply:
ok